code/espurna/garland/animations/anim_dolphins.h

```diff
@@ -51,25 +51,22 @@ class AnimDolphins : public Anim {
 
    private:
     struct Dolphin {
+        uint16_t numLeds;
         bool done = false;
-        int len = secureRandom(10, 20);
+        int len = secureRandom(10, 30);
         int speed = secureRandom(1, 3);
-        int dir = 1;
+        int dir = randDir();
         int head = 0;
         int start;
         Color color;
         std::unique_ptr<Color[]> points;
-        Dolphin(Palette* pal, uint16_t numLeds) : start(secureRandom(0, numLeds - len)), color(pal->getRndInterpColor()) {
+        Dolphin(Palette* pal, uint16_t numLeds) : numLeds(numLeds), start(head ? secureRandom(0, numLeds - len) : secureRandom(len, numLeds)), color(pal->getRndInterpColor()) {
```

Comment:
s/head/dir/?
(unless I am missing how it is anything but 0 from init above)

Reply:
Yes, it should be dir. Thanks!